lib/elastic_apm/util.rb

```diff
@@ -24,5 +24,12 @@ def self.hex_to_bits(str)
     def self.reverse_merge!(first, second)
       first.merge!(second) { |_, old, _| old }
     end
+
+    def self.truncate(value)
+      return unless value
+      return value if value.length <= 1024
+
+      format('%s…', value[0...1024])
```

Comment:
This won't work as you send 1024 + 3 chars for the dots, so the server will reject that. 

`format` creates a new copy here which I don't think is necessary. You could return `value[0,len]` or if you are keen on adding the dots something like `TRUNCATED_CONST="..."; value[0,len] << TRUNCATED_CONST`

Reply:
That’s a one-char utf ellipsis and a three-dot range operator. Means excluding the end. So actually 1024 chars. See the spec as proof 😊

Reply:
Your comment about allocating a copy is a valid point though 👍

Reply:
Ooh, I _was_ off by one – `value[0...(max_length - 1)] + '…'` checks out

Reply:
nitpick: the `s1 + s2` does also create a new string though..